src/include/function/copy/copy_function.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include "common/types/value/value.h"
+#include "function/function.h"
+
+namespace kuzu {
+namespace function {
+
+struct CopyFuncLocalState {
+    virtual ~CopyFuncLocalState() = default;
+
+    template<class TARGET>
+    TARGET& cast() {
+        return common::ku_dynamic_cast<CopyFuncLocalState&, TARGET&>(*this);
+    }
+};
+
+struct CopyFuncSharedState {
+    virtual ~CopyFuncSharedState() = default;
+
+    template<class TARGET>
+    TARGET& cast() {
+        return common::ku_dynamic_cast<CopyFuncSharedState&, TARGET&>(*this);
+    }
+};
+
+struct CopyFuncBindData {
+    std::vector<std::string> names;
+    std::vector<common::LogicalType> types;
+    std::string fileName;
+    bool canParallel;
```

Comment:
since we are materializing SERIAL now. Do we still have a case that we cannot parallelize COPY TO?